cluster/engine.go

```diff
@@ -470,8 +470,14 @@ func (e *Engine) updateClientVersionFromServer(serverVersion string) {
 		e.apiClient.UpdateClientVersion("1.23")
 	case versions.LessThan(serverVersion, "1.13"):
 		e.apiClient.UpdateClientVersion("1.24")
-	default:
+	case versions.LessThan(serverVersion, "1.13.1"):
 		e.apiClient.UpdateClientVersion("1.25")
+	case versions.LessThan(serverVersion, "17.03.1"):
```

Comment:
`version` is implemented here: https://github.com/moby/moby/blob/master/api/types/versions/compare.go#L10-L42
While I think it is better to strip off the suffix of `-ce` (maybe `-ee` as well, I have not experienced that yet).